azure-pipelines/main-pipeline.yml

```diff
@@ -163,62 +184,61 @@ stages:
 
       condition: eq(variables['${{ service.name }}'], 'True')
       steps:
+      
+      # install JDK 16
+      
+      - script: ./azure-pipelines/java-16-install.sh
+        name: Install_JDK_16
+      - task: JavaToolInstaller@0
+        inputs:
+          versionSpec: '16'
+          jdkArchitectureOption: 'x64'
+          jdkSourceOption: 'LocalDirectory'
+          jdkFile: './jdk-16.0.2_linux-x64_bin.tar.gz'
+          jdkDestinationDirectory: './jdk'
+          cleanDestinationDirectory: true
+
       - script: echo '${{ service.name }} has been updated - $(${{ service.name }})'
         displayName:  'Checking ${{ service.name }}'
 
       # Download Base Artifact to install in dependent services.
 
-      - task: DownloadPipelineArtifact@2
-        inputs:
-          buildType: 'current'
-          artifactName: 'base'
-          itemPattern: '**/*.jar'
-          targetPath: '$(Build.Repository.LocalPath)'
-
-
-      - task: SonarCloudPrepare@1
+      - task: UniversalPackages@0
+        displayName: Download Base Artifact
+        condition: eq('${{ service.requiresBase }}', 'True')
         inputs:
-          SonarCloud: 'SonarCloud'
-          organization: '$(sonarOrg)'
-          scannerMode: 'CLI'
-          configMode: 'manual'
-          cliProjectKey: '$(sonarProjectKey)'
-          cliProjectName: '$(sonarProjectName)'
-          cliSources: '.'
-          extraProperties: 'sonar.java.binaries=**/target/classes'
-
+          command: 'download'
+          downloadDirectory: '$(Build.Repository.LocalPath)'
+          feedsToUse: 'internal'
+          vstsFeed: $(artifactFeed)
+          vstsFeedPackage: $(artifactPackage)
+          vstsPackageVersion: '*'
 
       # Package Individual Service to create artifacts from.  
+      
+      - script: |
 
-      - script: |      
-
-          if [ ${{ service.requiresBase }} = true ]
+          if [ ${{ service.requiresBase }} = 'True' ]
```

Comment:
since we're using bash-specific syntax, this should really be a `bash` step rather than a `script` step so the pipeline is less platform-dependent. i won't block the PR for this though

Reply:
Got it.